common/components/common/upload/FileSelectButton.jsx

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import React from 'react';
+import {Button} from 'react-bootstrap';
+import FileDrop from 'react-file-drop';
+
+type Props = {|
+  onFileSelect: (image) => void,
+  buttonText: string,
+  dragText: string,
+  acceptedFileTypes: string,
+  iconClass: string,
+  hasImagePreview : boolean,
+  previewImage : string
+|};
+
+
+class FileSelectButton extends React.PureComponent<Props, State> {
+
+  constructor(): void {
+    super();
+    this.state = {
+    };
+  }
+
+  render(): React$Node {
+      const previewImageStyle = {
+        backgroundImage: 'url(' + this.props.previewImage + ')',
+    };
+
+      return (
+                <FileDrop onDrop={this._handleDrop.bind(this)} className={this.props.hasImagePreview ? "FileSelectButton-wrapper FileSelectButton-imagePreview" : "FileSelectButton-wrapper"}>
+                  <div className="FileSelectButton-innerWrapper" style={previewImageStyle} >
+                                <div className="FileSelectButton-innerContent">
+                    
+          <div className="FileSelectButton-dragFiles">{this.props.dragText || "Drag Your Files Here or"} </div>
+        <div className="FileSelectButton-button">
+           <input type="button" value={this.props.buttonText || "Browse Photos On Computer"} onClick={this._handleClick.bind(this)} className="btn btn-theme"/>
+          <input ref="fileInput" type="file" style={{display:"none"}} accept={this.props.acceptedFileTypes} onChange={this._handleFileSelection.bind(this)} />
+        </div>
+        </div>
+        </div>
+        </FileDrop>
+      );
+    
+  }
+
+   _processFile(file) : void {
+      const reader = new FileReader();
+      reader.addEventListener('load', () => {
+        this.setState({ src: reader.result });
+        this.props.onFileSelect(reader.result);
+      });
+      reader.readAsDataURL(file);
+    
+  };
+
+  _isValidFileType(file) : boolean {
+    // TODO check against this.props.acceptedFileTypes
```

Comment:
Are we okay with this waiting for a future pr?

Reply:
I think that's fine.  The default behavior of the react library seems to default to not letting users select non-image files.